src/steps/toolbx.rs

```diff
@@ -0,0 +1,74 @@
+use crate::error::TopgradeError;
+use anyhow::Result;
+
+use crate::executor::ExecutorOutput;
+use crate::terminal::print_separator;
+use crate::{execution_context::ExecutionContext, utils::require};
+use log::debug;
+use std::path::Path;
+use std::{
+    io::{self, Write},
+    process::Command,
+};
+
+fn list_toolboxes(toolbx: &Path) -> Result<Vec<String>> {
+    let output = Command::new(toolbx).args(&["list", "--containers"]).output()?;
+    let output_str = String::from_utf8(output.stdout)?;
+
+    let proc: Vec<String> = output_str
+        .lines()
+        // Skip the first line since that contains only status information
+        .skip(1)
+        .map(|line| match line.split_whitespace().nth(1) {
+            Some(word) => word.to_string(),
+            None => String::from(""),
+        })
+        .filter(|x| !x.is_empty())
+        .collect();
+
+    Ok(proc)
+}
+
+pub fn run_toolbx(ctx: &ExecutionContext) -> Result<()> {
+    print_separator("Toolbx");
+
+    let toolbx = require("toolbox")?;
+    let toolboxes = list_toolboxes(&toolbx)?;
+    debug!("Toolboxes to inspect: {:?}", toolboxes);
+
+    // Path of the running topgrade executable
+    let topgrade_path = String::from("/run/host") + std::env::current_exe()?.as_path().to_str().unwrap();
+
+    for tb in toolboxes.iter() {
+        print!("> Upgrading toolbox '{}'...", tb);
```

Comment:
Instead of capturing the output of the inner topgrade you can set the environment variable `TOPGRADE_PREFIX` to the name of the container and just run it as usual, Topgrade will print the prefix in each step.